# 01 - Data Collection

**Project: Credit Risk Prediction — Machine Learning-Based Credit Risk Assessment for Default Prediction in Consumer Lending**

This module loads the raw dataset and takes a first look at it: how many rows and columns, whether anything is missing, and what the target column (`Default`) looks like.

**Dataset:** `Loan_default.csv` — 255,347 real loan applications. Each row is one borrower.
`Default` = 1 means the borrower did not pay back the loan, 0 means they did.

**Input:** `data/Loan_default.csv`
**Output:** printed overview only (no files saved by this module)


In [1]:
import os
import pandas as pd

DATA_PATH = "data/Loan_default.csv"
TARGET = "Default"

pd.set_option("display.width", 120)
pd.set_option("display.max_columns", 50)

df = pd.read_csv(DATA_PATH)

print(f"Rows (borrowers)  : {df.shape[0]:,}")
print(f"Columns           : {df.shape[1]}")
print("\nFirst 5 rows:")
df.head()


Rows (borrowers)  : 255,347
Columns           : 18

First 5 rows:


,LoanID,Age,Income,LoanAmount,CreditScore,MonthsEmployed,NumCreditLines,InterestRate,LoanTerm,DTIRatio,Education,EmploymentType,MaritalStatus,HasMortgage,HasDependents,LoanPurpose,HasCoSigner,Default
0,I38PQUQS96,56,85994,50587,520,80,4,15.23,36,0.44,Bachelor's,Full-time,Divorced,Yes,Yes,Other,Yes,0
1,HPSK72WA7R,69,50432,124440,458,15,1,4.81,60,0.68,Master's,Full-time,Married,No,No,Other,Yes,0
2,C1OZ6DPJ8Y,46,84208,129188,451,26,3,21.17,24,0.31,Master's,Unemployed,Divorced,Yes,Yes,Auto,No,1
3,V2KKSFM3UN,32,31713,44799,743,0,3,7.07,24,0.23,High School,Full-time,Married,No,No,Business,No,0
4,EY08JDHTZP,60,20437,9139,633,8,4,6.51,48,0.73,Bachelor's,Unemployed,Divorced,No,Yes,Auto,No,0


In [2]:
print("Column names and their types:")
print(df.dtypes)

print("\nAny missing values?")
missing = df.isnull().sum()
missing = missing[missing > 0]
if len(missing):
    print(missing)
else:
    print("None - every cell in the dataset has a value.")

print(f"\nHow many borrowers defaulted?")
print(df[TARGET].value_counts())
print((df[TARGET].value_counts(normalize=True) * 100).round(2).astype(str) + "%")


Column names and their types:
LoanID                str
Age                 int64
Income              int64
LoanAmount          int64
CreditScore         int64
MonthsEmployed      int64
NumCreditLines      int64
InterestRate      float64
LoanTerm            int64
DTIRatio          float64
Education             str
EmploymentType        str
MaritalStatus         str
HasMortgage           str
HasDependents         str
LoanPurpose           str
HasCoSigner           str
Default             int64
dtype: object

Any missing values?
None - every cell in the dataset has a value.

How many borrowers defaulted?
Default
0    225694
1     29653
Name: count, dtype: int64
Default
0    88.39%
1    11.61%
Name: proportion, dtype: str


## How to read this

- We have **255,347 borrowers** and **18 columns** — plenty of data.
- **No missing values anywhere.** Real datasets usually have some, so this is lucky.
- Only **11.6% of borrowers defaulted** (about 1 in 9). Keep this figure in mind — it explains a lot of what comes later. A model that always guesses "no default" would already be 88% "accurate" without being useful, so accuracy alone will be misleading when we score models in Module 6.
- `LoanID` is a random ID per borrower and carries no risk information. It gets dropped in Module 3 (Data Preprocessing).

## Next module
Module 2 (Exploratory Analysis) looks at this same raw data more closely: averages, a correlation heatmap, and a couple of charts.
